onnxruntime/core/providers/cpu/generator/random.cc

```diff
@@ -363,9 +330,11 @@ static Status RandomUniformCompute(float low, float high,
 
 template <typename T, typename TDistribution>
 void GenerateData(std::default_random_engine& generator, TDistribution distribution, Tensor& tensor) {
-  auto out = gsl::make_span(tensor.template MutableData<T>(), tensor.Shape().Size());
-
-  std::for_each(out.begin(), out.end(), [&generator, &distribution](T& value) { value = distribution(generator); });
```

Comment:
gsl::span is good for bounds checking but inefficient for iteration so use a raw pointer instead as tensor.Shape() will guarantee correctness.